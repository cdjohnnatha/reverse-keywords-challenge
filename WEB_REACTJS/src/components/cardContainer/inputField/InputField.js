import React from 'react';
import { string, func } from 'prop-types';
import './InputField.css';

const InputField = ({ value, customStyle, onInputFieldChange }) => {

  let localStyle = {};
  if (customStyle) localStyle = { ...localStyle, customStyle };

  return (
    <input
      type="text"
      className="field input-style"
      value={value}
      onChange={onInputFieldChange}
      style={localStyle}
    />
  );
};

InputField.propTypes = {
  value: string,
  onInputFieldChange: func.isRequired,
};

InputField.defaultProps = {
  value: 'Kenna',
};

export default InputField;
