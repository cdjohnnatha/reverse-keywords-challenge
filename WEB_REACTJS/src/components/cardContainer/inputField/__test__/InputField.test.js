import React from 'react';
import { create } from 'react-test-renderer';
import '../../../../../jest.setup';
import InputField from '../InputField';

const changeWord = () => 'A';

describe('Test Inputfield component', () => {
  test('snapshot renders', () => {
    const component = create(<InputField onInputFieldChange={() => 'a'} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Inputfield component is a input element', () => {
    const component = create(
      <InputField value="Kenna" onInputFieldChange={() => changeWord('A')} />
    ).root;
    const inputField = component.findByType('input');

    expect(inputField.type).toBe('input');
    expect(inputField.props.value).toBe('Kenna');
  });
});
