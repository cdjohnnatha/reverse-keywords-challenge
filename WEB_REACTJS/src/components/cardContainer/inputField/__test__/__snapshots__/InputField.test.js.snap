// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Inputfield component snapshot renders 1`] = `
<input
  className="field input-style"
  onChange={[Function]}
  style={Object {}}
  type="text"
  value="Kenna"
/>
`;
